src/vuln_analysis/utils/chain_of_calls_retriever.py

```diff
@@ -271,6 +338,195 @@ def get_possible_docs(self, function_name_to_search: str, package: str, exclusio
 
         return [doc for doc in filter_1 if doc.page_content.__contains__(f"{function_name_to_search}(")]
 
+    def __find_caller_functions(self, document_function: Document, function_package: str) -> List[Document]:
```

Comment:
@RedTanny You don't have method overloading in python, then you selected to add s to the name.
But this not reflecting the function purposes, as you only returning one document, wrapped in a list.
maybe it's better to call it find_caller_function_bfs?